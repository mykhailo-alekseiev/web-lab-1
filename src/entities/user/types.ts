export enum Sex {
  Male = 'male',
  Female = 'female',
}

export interface GetUserInfoResponse {
  name: string
  sex: Sex
  email: string
}
