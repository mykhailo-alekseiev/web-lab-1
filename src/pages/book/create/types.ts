import { Book as BookType } from 'shared/types'

export type Book = Omit<BookType, 'createdAt' | '_id'>
