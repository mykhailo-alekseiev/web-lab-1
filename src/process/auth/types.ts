export interface SignUpRequest {
  email: string
  name: string
  password: string
  sex: string
}

export interface SignUpResponse {
  access_token: string
}

export interface SignInRequest {
  email: string
  password: string
}

export interface SignInResponse {
  access_token: string
}
