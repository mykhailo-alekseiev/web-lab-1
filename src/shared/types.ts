export enum BookCategory {
  Adventure = 'adventure',
  Classic = 'classic',
  Crime = 'crime',
  Love = 'love',
}

export interface Book {
  _id: string
  title: string
  description: string
  author: string
  price: number
  category: BookCategory
  createdAt: Date
}
