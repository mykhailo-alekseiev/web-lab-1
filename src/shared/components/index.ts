export * from './loadable'
export * from './loader'
export * from './route-guard/guest-guard'
export * from './route-guard/auth-guard'
export * from './not-found-page-redirect'
