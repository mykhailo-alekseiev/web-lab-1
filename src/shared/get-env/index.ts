/**
 * Initialization of env variables module
 * @remark If the value of any variable is not found,
 * the application will immediately throw an error when initializing the module
 * @module
 */

/**
 * Getting an env variable
 * @throwable
 */
const getEnvVar = (key: string) => {
  if (!process.env[key]) {
    throw new Error(`Env variable ${key} is required`)
  }

  return process.env[key] || ''
}

export const REACT_APP_SERVER_URL = getEnvVar('REACT_APP_SERVER_URL')
