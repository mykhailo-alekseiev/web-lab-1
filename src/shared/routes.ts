export const Page = {
  Home: '/',
  SignUp: '/auth/sign-up',
  SignIn: '/auth/sign-in',
  CreateBook: '/book/create',
  UserProfile: '/user/profile',
}
