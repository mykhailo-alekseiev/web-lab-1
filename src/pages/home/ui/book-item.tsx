import React, { FC } from 'react'
import { Card, CardContent, Divider, Stack, Typography } from '@mui/material'
import { DeleteBookButton } from 'features/book/delete'
import dayjs from 'dayjs'
import { Book } from 'shared/types'

type Props = Book

const BookItem: FC<Props> = ({ _id, title, createdAt, description, author, category, price }) => {
  return (
    <Card key={_id}>
      <CardContent>
        <Stack spacing={1}>
          <Stack spacing={1} direction='row' justifyContent='space-between'>
            <Typography color='text.secondary' variant='h5' fontWeight={700}>
              Title: {title}
            </Typography>
            <DeleteBookButton bookId={_id} />
          </Stack>
          <Typography color='text.secondary'>Description: {description}</Typography>
          <Stack direction='row' spacing={2} divider={<Divider orientation='vertical' flexItem />}>
            <Typography color='text.secondary'>Author: {author}</Typography>
            <Typography color='text.secondary'>Categody: {category}</Typography>
            <Typography color='text.secondary'>Price: {price}</Typography>
            <Typography color='text.secondary'>
              Created at: {dayjs(createdAt).format('MM/DD/YYYY')}
            </Typography>
          </Stack>
        </Stack>
      </CardContent>
    </Card>
  )
}

export { BookItem }
