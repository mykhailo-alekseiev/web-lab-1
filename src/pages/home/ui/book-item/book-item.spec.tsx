import { cleanup, render } from '@testing-library/react'
import { BookItem } from 'pages/home/ui/book-item/index'
import React from 'react'
import { BookCategory } from 'shared/types'
import '@testing-library/jest-dom'
import dayjs from 'dayjs'

afterEach(cleanup)

describe('book item', () => {
  it('correct display', () => {
    const createdAt = new Date()

    const { asFragment, container } = render(
      <BookItem
        _id='random'
        title='Title1'
        description='Description 1'
        author='Ryan Gosling'
        price={100}
        category={BookCategory.Classic}
        createdAt={createdAt}
      />,
    )

    expect(container).toHaveTextContent('Title: Title1')
    expect(container).toHaveTextContent('Description: Description 1')
    expect(container).toHaveTextContent('Author: Ryan Gosling')
    expect(container).toHaveTextContent(`Category: ${BookCategory.Classic}`)
    expect(container).toHaveTextContent('Price: 100')
    expect(container).toHaveTextContent(`Created at: ${dayjs(createdAt).format('MM/DD/YYYY')}`)

    expect(container.querySelector('div')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
