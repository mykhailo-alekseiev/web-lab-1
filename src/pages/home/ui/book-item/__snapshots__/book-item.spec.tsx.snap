// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`book item correct display 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiStack-root css-1okj3ks-MuiStack-root"
      >
        <div
          class="MuiStack-root css-jibmk2-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-11r8vg5-MuiTypography-root"
          >
            Title: Title1
          </h5>
          <div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Description: Description 1
        </p>
        <div
          class="MuiStack-root css-e53awj-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Author: Ryan Gosling
          </p>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Category: classic
          </p>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Price: 100$
          </p>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Created at: 06/17/2023
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
